compiler-rt/lib/fuzzer/FuzzerUtilFuchsia.cpp

```diff
@@ -607,7 +607,14 @@ size_t PageSize() {
 }
 
 void SetThreadName(std::thread &thread, const std::string &name) {
-  // TODO ?
+  std::string tmp(name);
+  if (tmp.resize() > 31)
+    tmp.resize(31);
+  if (zx_status_t s = zx_object_set_property(
+          thread.native_handle(), ZX_PROP_NAME, tmp.c_str(), tmp.size());
+      s != ZX_OK)
```

Comment:
I talked to the kernel folks. The ZX_PROP_NAME doesn't need the string to be a NULL-terminated and it doesn't need to be size capped. The kernel will copy the string, NULL-terminate it and cap the size for you. In this case, you can just do 

```
zx_status_t s = zx_object_set_property(thread.native_handle(), ZX_PROP_NAME, name.data(), name.size())
```
And remove the local tmp variable to avoid a new string allocation.

Reply:
ah good to know !